doc/mesh-adapter-api.txt

```diff
@@ -0,0 +1,137 @@
+BlueZ D-Bus Mesh Adapter API description
+********************************
+
+Mesh Network Hierarchy
+======================
+Service		org.bluez.mesh
+Interface	org.bluez.mesh.AdapterManager1
+Object path	/org/bluez/mesh/adapter
+
+Methods:
+	uint64 token Add(string io_type, dict config)
+
+	    This is method that an application has to call to
+	    add a new mesh adapter with an IO type.
+
+	    The config parameter is a dictionary with configuration
+	    options specific for the chosen IO type. In case of "tcpserver",
+	    mesh adapter with given UUID and NetworkKey can added only once.
+	    When adapter with UUID and NetworkKey is already exists then
+	    AlreadyExists error will be raised. When one of required
+	    configuration options is missing then InvalidArguments
+	    error will be raised.
+
+        string io_type
+
+            This parameter choose which IO type will be used
+            to communicate with an adapter.
+
+            The defined values are: "tcpserver", "silvair"
+
+        dict config
+
+            Options available for the "tcpserver" IO:
+
+                int Port
+
+                    Port for listening for incoming connections.
+                    This field is mandatory.
```

Comment:
Is this port on `meshd` (server) side or uGW (client)? If it's the former, then the daemon should already know it. Or does this option turn the TCP server on?

Reply:
It is `meshd` (server) side port. You are right it can be defined during daemon start.

Reply:
It's the server side, so it might make sense to make this optional with some default.

Also, we're missing the list of interfaces (IP addresses) the server should bind to - at the moment we bind to `INADDR_ANY`, but this should be configurable.